src/backend/langflow/components/vectorstores/Weaviate.py

```diff
@@ -0,0 +1,79 @@
+import weaviate # type: ignore
+from typing import Optional, Union
+from langflow import CustomComponent
+
+from langchain.vectorstores import Weaviate
+from langchain.schema import Document
+from langchain.vectorstores.base import VectorStore
+from langchain.schema import BaseRetriever
+from langchain.embeddings.base import Embeddings
+
+
+class WeaviateVectorStore(CustomComponent):
+    display_name: str = "Weaviate"
+    description: str = "Implementation of Vector Store using Weaviate"
+    documentation = (
+        "https://python.langchain.com/docs/integrations/vectorstores/weaviate"
+    )
+    beta = True
+    field_config = {
+        "url": {"display_name": "Weaviate URL", "value": "http://localhost:8080"},
+        "api_key": { "display_name": "API Key", "password": True,"required": False, },
+        "index_name": {"display_name": "Index name","required": False,},
+        "text_key": {"display_name": "Text Key","required": False,  "advanced": True, "value": "text"},
+        "documents": {"display_name": "Documents", "is_list": True},
+        "embedding": {"display_name": "Embedding"},
+        "attributes": {"display_name": "Attributes", "required": False, "is_list": True, "field_type": "str", "advanced": True},
+        "search_by_text": {"display_name": "Search By Text", "field_type": "bool", "advanced": True},
+        "code": {"show": False},
+    }
+
+    def build(
+        self,
+        url: str,
+        search_by_text: bool = False,
+        api_key: Optional[str] = None,
+        index_name: Optional[str] = None,
+        text_key: Optional[str] = "text",
+        embedding: Optional[Embeddings] = None,
+        documents: Optional[Document] = None,
+        attributes: Optional[list] = None,
+    ) -> Union[VectorStore, BaseRetriever]:
```

Comment:
@ogabrielluiz btw, this component can be used also as a `BaseRetriever`, in case when the embeddings are already stored in Weaviate index. 

For example, I have several ETL jobs that load data from multiple sources and store the embeddings in Weaviate (done with Llamaindex). Using Langflow, I can retrieve the data from the index.

Q: Should I split this into 2 components? `VectorStore` + `BaseRetriever`?

Reply:
No. Most of the VectorStores can be both. Have you tested this component as both?

Reply:
yes!